sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SpannerMutationCoder.java

```diff
@@ -0,0 +1,224 @@
+package org.apache.beam.sdk.coders;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Date;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Timestamp;
+import com.google.cloud.spanner.Type;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.spanner.ValueBinder;
+import com.google.common.io.ByteArrayDataInput;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+/**
+ * A {@link Coder} that encodes Spanner {@link Mutation} objects.
+ */
+public class SpannerMutationCoder extends AtomicCoder<Mutation> {
```

Comment:
Yes.   This is the reason for the custom Codec.   If Mutation implements the protobuf Message interface then we can deprecate these.   I will write unit tests for MutationCoder and StructCoder if necessary.   These classes need to be optimized for minimal write output.  Right now type names serialize as strings.